src/bsp/src/stm32/src/interloc/src/TwoWayRanging.cpp

```diff
@@ -24,19 +24,19 @@ void TwoWayRanging::constructFinal(UWBMessages::TWRFinal* finalMessage, uint64_t
     DecawaveUtils::tsToBytes((uint8_t*)(&finalMessage->m_finalTxTs), m_finalTxTs);
 }
 
-std::optional<double> TwoWayRanging::calculateDistance() const {
-    uint64_t tRound1 = m_responseRxTs[RESPONDER_SLOT - 1] - m_pollTxTs;
+std::optional<double> TwoWayRanging::calculateDistance(uint16_t slotId) const {
+    uint64_t tRound1 = m_responseRxTs[slotId - 1] - m_pollTxTs;
     uint32_t tRound2 = (uint32_t)m_finalRxTs - (uint32_t)m_responseTxTs;
     uint32_t tReply1 = (uint32_t)m_responseTxTs - (uint32_t)m_pollRxTs;
-    uint64_t tReply2 = m_finalTxTs - m_responseRxTs[RESPONDER_SLOT - 1];
+    uint64_t tReply2 = m_finalTxTs - m_responseRxTs[slotId - 1];
 
     uint64_t tofDtu =
         (tRound1 * tRound2 - tReply1 * tReply2) / (tRound1 + tRound2 + tReply1 + tReply2);
 
     double tof = tofDtu * DWT_TIME_UNITS;
     double distance = tof * SPEED_OF_LIGHT;
 
-    if (distance > 0 && distance < MAX_POSSIBLE_DISTANCE) {
+    if (distance < MAX_POSSIBLE_DISTANCE) {
```

Comment:
what about distance < 0?

Reply:
tof is a uint, so it can never be <0, but I will readd the >0 check to prevent returning 0